### ISSUE TITLE
About the inference process

### ISSUE THREAD

**Issue by gmftbyGMFTBY** [andrewpeng02/transformer-translation#1]
Hi, thanks for your wonderful tutorial for the Pytorch nn.Transformer.

I'm a little bit confused about your codes in the file `translate-sentence.py`.
In this file, you generate the output by this line:

```python
output = model.forward(src, tgt, src_key_padding_mask=None, tgt_key_padding_mask=None, memory_key_padding_mask=None, tgt_mask=tgt_mask)
```

In this part of codes, in my view, I think the `src_key_padding_mask` and `memory_key_padding_mask` are also very necessary. So why did you set the `None` to these two parameters? Can you explain it to me?

Thank you so much and hope to get the responses from you.

**Comment by andrewpeng02** [comment]
Hi, I'm glad you found this repository helpful. 

In the inference process I set ``src_key_padding_mask``, ``tgt_key_padding_mask``, and ``memory_key_padding_mask`` to ``None`` because I'm passing only one sentence in the model at a time. There's no need for padding because I'm only sending 1 source and target sentence.
In the training process I do need those padding_masks because I have a ``batch_size`` > 1 and I need to add ``[PAD]`` tokens up to the maximum sentence length.

**Comment by gmftbyGMFTBY** [comment]
Wow, thank you for your response. 
Understand.  Close the issue! Thank you for your help 🤗

**Comment by MCLYang** [comment]
Hi, thank you for your project and post. You code is most understandable I have ever found in git. I am trying to understand for the forward process for `translate-sentence.py` as well.
```
    # Generate the translated sentence, feeding the model's output into its input
    translated_sentence = [fr_freq_list['[SOS]']]
    i = 0
asdasdasd        output = forward_model(model, sentence, translated_sentence).to('cuda')
        values, indices = torch.topk(output, 5)
        translated_sentence.append(int(indices[-1][0]))
```
what is the `[fr_freq_list['[SOS]']]` you put at the first place? I mean the first time pass to `forward_model`? Is it just Blank? 

Sorry I did not run your code because this is only place that confuse me. 

Thank you
